src/me/mrCookieSlime/Slimefun/Objects/MultiBlock.java

```diff
@@ -12,18 +13,56 @@
 public class MultiBlock {
 	
 	private Material[] blocks;
-	private Material trigger;
-	
+	private BlockFace trigger;
+	private boolean isSymmetric;
+
+	@Deprecated
 	public MultiBlock(Material[] build, Material trigger) {
+		this.blocks = build;
+		this.isSymmetric = isSymmetric(build);
+		//Hacky, the other constructor should be used
+		for (int i = 0; i < 9; i++)
+		{
+			if (trigger.equals(build[i]))
+			{
+				switch (i) {
```

Comment:
Do we really need a switch/case here for free options?
I mean, 3 if-statements would do the trick as well.

Reply:
since this is a numeric lookup I support the use of a switch here, allows for much better JVM optimisation

Reply:
Okay then

Reply:
Thats up to the coding style/preferences. If you prefer three if statements i can change that, i personally prefer switch statements.